### ISSUE TITLE
Combination of Dependency Injection and custom handler is not detected

### ISSUE THREAD

**Issue by maschnetwork** [aws/aws-lambda-snapstart-java-rules#28]
Hi,

thanks for fixing the other issues. However, I see that a combination of the issues I reported are not recognized. For example, having a Functional interface (#15 ) with Dependency Injection (#14 ) is not recognized:

```
public class DependencyInjection implements Function<String, String> {
    private final MyDependency myDependency;

    public DependencyInjection(MyDependency myDependency) {
        this.myDependency = myDependency;
    }

    @Override
    public String apply(String s) {
        return "Stuff";
    }
}

```

```
public class MyDependency {
    private final UUID random = UUID.randomUUID();

    public UUID getUUID() {
        return random;
    }
}

```

If I change the DependencyInjection to a RequestHandler Interface it works.

Best,
Max

**Comment by mbfreder** [comment]
Hi, thanks for the quick feedback. We are working to fix this.
